### ISSUE TITLE
Bundle Identifier should not be an array

### ISSUE THREAD

**Issue by devboy** [fastlane/fastlane#11324]
For some reason the bundle identifier of my app is turned into an array at some point...

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

##### Complete output when running fastlane, including the stack trace and command used
```
$ fastlane match
...
[!] Couldn't find bundle identifier '["com.xyz"]' for the user 'me@you.org'
```
### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.2                                     |
| Ruby                        | 2.4.2                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.15.0                          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.13.2 (17C88)                    |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.2_1/lib          |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.70.3  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.3        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.3        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.4.5        |
| plist                     | 3.1.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| tty-screen                | 0.6.3        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.54.0       |
| faraday                   | 0.9.2        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.2        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.16.0       |
| faraday_middleware        | 0.10.0       |
| uber                      | 0.0.15       |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| json                      | 2.0.4        |
| io-console                | 0.4.6        |
| openssl                   | 2.0.5        |
| rubyzip                   | 1.2.1        |
| mini_portile2             | 2.2.0        |
| nokogiri                  | 1.8.0        |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
</details>

*generated on:* **2017-12-22**
</details>

**Comment by KrauseFx** [comment]
Yah, under the hood it's handled as an array to iterate over it. Can you share your configuration files so we can take a look? 
Also we need the complete output of when you run it, in particular the summary table when running _match_. Thanks 👍

**Comment by dougsuriano** [comment]
Also seeing errors regarding identifying build identifiers when using `match` ... did a little debugging.

I think this change 36b56785a6bebfa54ee727ac170228f0eb4ce24d is causing the [Match runner](https://github.com/fastlane/fastlane/blob/master/match/lib/match/runner.rb#L28) to always think the `app_identifier` is an array. As a result, when passing in only 1 bundle identifier, `Spaceship.app.find` checks if `["your_bundle_id"]` exists versus `"your_bundle_id"`.

**Comment by taquitos** [comment]
I'm taking a look at this, it was supposed to be fixed in that commit @dougsuriano 😭

**Comment by dougsuriano** [comment]
@taquitos No worries! Lemme know if I can be of any help. Happy Holidays!

**Comment by taquitos** [comment]
@dougsuriano thank you, you too!

**Comment by KrauseFx** [comment]
We shipped https://github.com/fastlane/fastlane/pull/11212 a few days ago, are you on the latest fastlane release?

**Comment by taquitos** [comment]
Ok, I'm going to submit a patch that should fix this, it's just temporary though. A better fix is to address `ConfigItem`s that have parameters with multiple `type`s , for example, 'app_identifier' can be a `String` or `Array`.

I want to unblock folks though, so I'm noting it in the code and on this issue. I wouldn't normally do this since I believe in fixing it right, the first time, but I just don't want to leave people hanging.
